design/Proposal for View Schema.md

```diff
@@ -0,0 +1,57 @@
+﻿
+
+
+
+## Schema definition for a View
+This is proposed  specification for a  schema definition of a view.  This is based on what a Dremio view schema would need to save along with the SQL text string describing the view. The goal is this schema could be used generally as well. 
+Described is an outline of the classes for the types and attributes for the view fields. 
+
+(This schema is a data member of the DremioView definition )
+
+A ViewSchema is a list of ViewFields and maintains a versionId to handle future version changes if any.
+
+    public final class ViewSchema implements Externalizable, Message<ViewSchema>, Schema<ViewSchema>  {      
+        private Integer versionId;  
+        private List<ViewField> field;
+     }
+A ViewField describes the attributes of the field - Simple, Struct, List, Map types are covered.
+
+    public final class ViewField implements  Externalizable, Message<ViewField>, Schema<ViewField> {    
```

Comment:
As previously mentioned can we copy someone elses format? Arrow is probably the most complete but Parquet, Iceberg and potentially Hive are geared towards SQL.

Hive has a simple thrift format for its schemas (though I can't find a list of data types)
Iceberg has a well defined schema as well: http://iceberg.apache.org/spec/#schemas-and-data-types